harness/tests/launch/test_wrap_rank.py

```diff
@@ -0,0 +1,50 @@
+import subprocess
+import sys
+import textwrap
+
+from determined.launch import wrap_rank
+
+
+def test_split_on_new_lines_or_carriage_returns() -> None:
+    script = textwrap.dedent(
+        r"""
+        import sys
+        print("line with lf", end="\n", flush=True)
+        print("check lf", file=sys.stderr, flush=True)
+        input()
+        print("line with cr", end="\r", flush=True)
+        print("check cr", file=sys.stderr, flush=True)
+        input()
+    """
+    )
+    cmd = [
+        sys.executable,
+        "-u",
+        wrap_rank.__file__,
+        "--no-redirect-stdio",
+        "0",
+        sys.executable,
+        "-u",
+        "-c",
+        script,
+    ]
+    p = subprocess.Popen(
+        cmd,
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    assert p.stdin and p.stdout and p.stderr
```

Comment:
The more assertions, the less it's clear to me what's under test. Short chat?

Reply:
discussed offline, added a comment to the mypy assertion and agreed that the p.wait()==0 is just about being strict, not truly its own test.